### PR TITLE
ci: add node version matrix

### LINKED ISSUE #159: ci: add matrix of supported node versions
Check https://github.com/vitejs/vite/blob/main/.github/workflows/ci.yml#L27 how to do it

### PR DESCRIPTION
**Pull request by MohdImran001**
Fixes #159 

@prisis please review

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -7,6 +7,18 @@ on:
   pull_request:
 
 jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+        node_version: [12, 14, 16, 17]
```
Comment (Shinigami92) [inline .github/workflows/ci.yml:15]: You need to reuse the `node_version` down below in `actions/setup-node@v2`

**Review by Shinigami92 — CHANGES_REQUESTED** [review]

**Comment by prisis** [comment]
@MohdImran001 do you need help with it?

**Comment by MohdImran001** [comment]
Sorry I was a bit busy , I will complete this today.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -19,27 +19,45 @@ jobs:
...
     steps:
-      - uses: actions/checkout@v2
-      - uses: actions/setup-node@v2
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Set node version to 16
+        uses: actions/setup-node@v2
         with:
           node-version: '16'
```
Comment (Shinigami92) [inline .github/workflows/ci.yml:53]:
```suggestion
          node-version: 16
```

**Review by Shinigami92 — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -1,4 +1,4 @@
-name: 'Continuous Integration'
+name: 'CI'
```
[the remark this answers is not in the document]
Comment (MohdImran001) [reply .github/workflows/ci.yml]: already done

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -19,27 +19,45 @@ jobs:
...
-      - run: npm ci
-      - run: npm run build
-      - run: npm run lint
+
+      - name: Install deps
+        run: npm ci
+
+      - name: Build
+        run: npm run build
+
```
[the remark this answers is not in the document]
Comment (MohdImran001) [reply .github/workflows/ci.yml:61]: remove it from both testing and linting ?

[the remark this answers is not in the document]
Comment (Shinigami92) [reply .github/workflows/ci.yml:61]: No only linting ^^

**Comment by Shinigami92** [comment]
We need to find out what the `npm ERR! google-closure-compiler-linux not accessible from google-closure-compiler`-error is
And why it fails with one test in windows :thinking:

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -8,26 +8,53 @@ on:
...
-    runs-on: ubuntu-latest
-    name: 'Testing'
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+        node_version: [12, 14, 16, 17]
+        include:
+          - os: macos-latest
+            node_version: 16
```
Comment (poyea) [inline .github/workflows/ci.yml:18]:
```suggestion
            node_version: '16.x'
```

Comment (poyea) [reply .github/workflows/ci.yml:18]: [Why this](https://github.com/faker-js/faker/blob/main/test/finance.unit.js#L181) is [taking so long](https://github.com/faker-js/faker/runs/4856849312?check_suite_focus=true#step:6:23415) on Windows?

Comment (Shinigami92) [reply .github/workflows/ci.yml:18]: > node_version: '16.x'

Why would this be needed? Theoretically it's the same
So I would suggest to not do that

**Comment by Shinigami92** [comment]
@MohdImran001 Please remove the dependency `"node-minify": "*",` and then run `rm -Rf node_modules package-lock.json && npm install`
Then commit both files: `package.json` and `packge-lock.json`

File: .github/workflows/ci.yml (hunk 5 above)
Comment (poyea) [reply .github/workflows/ci.yml:18]: Indeed, I believe this is true due to semantic versioning. Well, I think this doesn't matter much, so it's OK

Comment (poyea) [reply .github/workflows/ci.yml:18]: Btw, [this or similar options](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepstimeout-minutes) might be useful, I didn't look into the Windows issue though

**Comment by Shinigami92** [comment]
Nice, now node 14 and 12 is working :+1: 
But we need to find out why the one test in windows is failing

**Comment by Shinigami92** [comment]
It's actually this test:

https://github.com/faker-js/faker/blob/d72482b0a19c3e98b6c8ea96d6c8407711af60cc/test/finance.unit.js#L283-L299

**Comment by ST-DDT** [comment]
I am unable to reproduce the test error on both my win11 and win10 computers.

**Comment by Shinigami92** [comment]
I hacked around it by conditionally disable the test on `win32`
At least it is better for now to have the other machines running all the tests, so in total we have 4x22k+ more tests running on node 12, 14 and mac and win
I will create an issue so we remember to fix this test later

**Review by Shinigami92 — APPROVED** [review]

**Review by ST-DDT — APPROVED** [review]

**Review by griest024 — APPROVED** [review]

**Pull request merged**